packages/form-js-viewer/src/core/PathRegistry.js

```diff
@@ -0,0 +1,179 @@
+import { isArray } from 'min-dash';
+import { clone } from '../util';
+
+export default class PathRegistry {
```

Comment:
What can help a lot I guess is to add some documentation here about the whole purpose of the path registry module and the mental model behind it. This will help in some other places where the complexity of the nested data kicks in place (dragging, importer, ...)